app/src/main/java/com/example/egida/presentation/ui/SettingFragment.kt

```diff
@@ -22,20 +23,109 @@ class SettingFragment : Fragment() {
         const val TAG = " SettingFragment"
     }
 
-    private lateinit var mBinding: SettingFragmentBinding
-    private lateinit var viewModel: SettingViewModel
+    private lateinit var binding: SettingFragmentBinding
+    private lateinit var settingViewModel: SettingViewModel
     private lateinit var mainViewModel: MainViewModel
 
 
     override fun onStart() {
         super.onStart()
         mainViewModel.closeDrawer(requireActivity())
-        Log.d(FitFragment.TAG, " $DAY")
     }
 
     override fun onResume() {
         super.onResume()
-        changeValuesFragmentFields()
+        binding.editFirstName.doAfterTextChanged { editable ->
+            lifecycleScope.launch {
+                settingViewModel.userDatabase
+                    .collect { userDatabase ->
+                        userDatabase.firstName = editable.toString()
+                        Log.d(TAG, userDatabase.firstName)
```

Comment:
You should not mutate data from UI. Here call 'viewModel.setUserName(editable.toString). And no need in coroutines.